src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java

```diff
@@ -63,14 +49,26 @@ public UserServiceImpl(
         this.emailContent = emailContent;
     }
 
+    public User getLoggedUser() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String currentUser = authentication.getName();
+
+        return userRepository.findOneByEmail(currentUser).orElseThrow(() ->
+                new NotLoggedInException("You are not logged in!"));
+    }
+
     @Override
     public List<User> getAll() {
         return userRepository.findAll();
     }
 
     @Override
-    public List<User> saveAll(Collection<User> users) {
-        return userRepository.saveAll(users);
+    public void saveAll(Collection<User> users) {
+        userRepository.saveAll(users);
+    }
+
+    public List<FriendDetails> getUserFriendsSortedByLocation(Long userId, FriendSearchDto friendSearchDto) {
+        return userRepository.findUserFriendsSortedByLocation(userId, friendSearchDto.getCurrentLatitude(), friendSearchDto.getCurrentLongitude());
```

Comment:
Where is the sort option here? I dont see a way to sort the returned results, maybe the sorting is hardcoded in the query? Can we pass a sort parameter here?

Reply:
The task is to sort them by location i don't have options to sort them in any other way. And I am using a formula for the sorting, because I am getting their location as coordinates. I used the Query to save writing more methods.

Reply:
Okay, don't bother with that. Review the other suggestions. I agree with your arguments.